ui-tests/src/test/java/org/openmrs/reference/AddEditDeleteFormTest.java

```diff
@@ -2,57 +2,79 @@
 
 import org.junit.After;
 import org.junit.Before;
-import org.junit.Ignore;
 import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.openmrs.reference.groups.BuildTests;
+import org.openmrs.reference.page.AdministrationPage;
 import org.openmrs.reference.page.ClinicianFacingPatientDashboardPage;
 import org.openmrs.reference.page.HeaderPage;
+import org.openmrs.reference.page.HtmlFormsPage;
+import org.openmrs.reference.page.ManageHtmlFormsPage;
 import org.openmrs.reference.page.HomePage;
 import org.openmrs.reference.page.ManageFormsPage;
+import org.openqa.selenium.By;
 import org.openmrs.uitestframework.test.TestBase;
 
 import static org.junit.Assert.assertNotNull;
 
-
 /**
  * Created by nata on 24.06.15.
  */
-public class EditFormTest extends TestBase {
+public class AddEditDeleteFormTest extends TestBase {
+
+    private static String name = "newFormTest1";
+    private static String description = "description of new form";
+    private static String version = "1.2";
     private HomePage homePage;
+    private AdministrationPage administrationPage;
     private HeaderPage headerPage;
     private ManageFormsPage manageForm;
+    private ManageHtmlFormsPage manageHtmlFormsPage;
+    private HtmlFormsPage htmlFormsPage;
     private ClinicianFacingPatientDashboardPage patientDashboardPage;
 
     @Before
     public void setUp() throws Exception {
-
         homePage = new HomePage(page);
-        assertPage(homePage);
+        assertPage(homePage.waitForPage());
         headerPage = new HeaderPage(driver);
+        administrationPage = new AdministrationPage(page);
         manageForm = new ManageFormsPage(driver);
+        htmlFormsPage = new HtmlFormsPage(page);
+        manageHtmlFormsPage = new ManageHtmlFormsPage(page);
         patientDashboardPage = new ClinicianFacingPatientDashboardPage(page);
     }
 
-    @Ignore//ignore due to moving forms functionality
     @Test
-    public void EditFormTest() throws Exception {
+    @Category(BuildTests.class)
+    public void addFormTest() throws Exception {
+
+        manageHtmlFormsPage = homePage.goToAdministration().clickOnManageHtmlForms();
+        if (manageHtmlFormsPage.getElementsIfExisting(By.xpath("//*[contains(text(), '" + name + "')]")).isEmpty()) {
+            manageHtmlFormsPage.clickOnNewHtmlForm();
+            htmlFormsPage.createNewFormTest(name, description, version);
+        }
+        homePage.go();
         homePage.goToManageForm();
         if (!manageForm.addPresent()) {
             manageForm.delete();
         }
+        // add Form
```

Comment:
each add/edit/delete should rather be a separate test method and not all in one

Reply:
Let me separate them into methods

Reply:
You may also need to rename the class as FormTest